src/main/java/com/iota/iri/zmq/TimeWindowedApproveeCounter.java

```diff
@@ -0,0 +1,93 @@
+package com.iota.iri.zmq;
+
+import com.iota.iri.controllers.TransactionViewModel;
+import com.iota.iri.model.Hash;
+import com.iota.iri.storage.Tangle;
+import com.iota.iri.utils.dag.DAGHelper;
+import com.iota.iri.utils.dag.TraversalException;
+
+import java.util.HashSet;
+
+/**
+ * Utility class to count the number of distinct approvees that have an arrival time within the given time window.
+ */
+class TimeWindowedApproveeCounter {
+
+    private final Tangle tangle;
+    private final long minTransactionAgeSeconds;
+    private final long maxTransactionAgeSeconds;
```

Comment:
Why not `java.time.Duration`? You start there anyway

Reply:
Migrated all the arguments to Duration and Instant to have time units. However, I kept the fields in seconds as the `transaction.getArrivalTime()` is also in seconds and converting them every time did not improve readability.